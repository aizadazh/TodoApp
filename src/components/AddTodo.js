import React from 'react';


const AddTodo = ({ submitTodo, undeleteTodo, inputChanged, disableAddTodo, disableUndelete }) => {
  let input;

  return (
    <div>
      <form
        onSubmit={(event) => {
          event.preventDefault();
          submitTodo(input.value);
          input.value = '';
        }}
      >

        <input
          className="todo-input"
          ref={(element) => {
            input = element;
          }}
          onChange={() => inputChanged(input.value)}
          placeholder="I'm going to..."
        />

        <button
          type="submit"
          className="todo-submit"
          disabled={disableAddTodo}
        >
          Add Todo
        </button>

        <button
          className="todo-undelete"
          onClick={() => undeleteTodo()}
          disabled={disableUndelete}
        >
          Undelete
        </button>

      </form>
    </div>
  );
};


export default AddTodo;