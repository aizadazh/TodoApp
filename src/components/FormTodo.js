import React from 'react';
function FormTodo(props) {
    let input;
    return (
        <form
            onSubmit={(event) => {
                event.preventDefault();
                props.editTodo(input.value, props.todo.id);
            }}
        >
            <div style={{display:props.todo.show}}>
                <input
                    className="todo-input"
                    ref={(element) => {
                        input = element;
                    }}
                    onChange={() => props.inputChanged(input.value)}
                    placeholder={props.todo.text}
                />
                <button
                    type="submit"
                    className="todo-submit"
                    disabled={props.disableAddTodo}
                >
                    Submit
          </button>
            </div>

        </form>

    )

};


export default FormTodo;
