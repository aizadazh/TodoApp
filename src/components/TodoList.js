import React from 'react';
import FormTodo from './FormTodo'

const TodoList = ({ todos, deleteTodo, inputChanged, editTodo, disableAddTodo, show}) => {
  
  if (todos) {
    var todoItems = todos.map(todo => {
      return (  
        <li key={todo.id}>     
          <button
            type="button"
            className="todo-delete"
            onClick={() => deleteTodo(todo.id)}
          >Delete</button>
          <button
            className="todo-submit"
            onClick={()=> show(todo.id)}
          >
            Edit
          </button>
          
          <span className="todo-text">
            {todo.text}
          </span>
              <FormTodo todo={todo} inputChanged={inputChanged} editTodo={editTodo} disableAddTodo={disableAddTodo} />
        </li>

      )
    });
  }


  return (

    <ol>
      {todoItems}
    </ol>
  );
};






export default TodoList;