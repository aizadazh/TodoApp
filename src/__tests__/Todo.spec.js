import React from "react";
import { shallow } from "enzyme";
import { Todo } from "../components/Todo";

describe('Testing Todo Component', () => {
    const props = {
        todo: jest.fn(),
        toggleTodo: jest.fn()
    }
   
    const wrapper = shallow(<Todo {...props} />)
        it('renders correctly', ()=>{
            expect(wrapper).toMatchSnapshot();  
            expect(wrapper.find('li').hasClass('todo-item')).toBe(true);
        })
        it('handles onClick', () => {
            const onClick = jest.fn();
            const wrapper = shallow(<li onClick={onClick} />)
            wrapper.find('li').simulate('click')
            expect(onClick).toMatchSnapshot();
        }) 
    })       


