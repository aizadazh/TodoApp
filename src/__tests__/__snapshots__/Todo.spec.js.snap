// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Todo Component handles onClick 1`] = `
[MockFunction] {
  "calls": Array [
    Array [],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`Testing Todo Component renders correctly 1`] = `ShallowWrapper {}`;
